lib/travis/build/templates/header.sh

```diff
@@ -265,6 +265,37 @@ travis_fold() {
   echo -en "travis_fold:${action}:${name}\r${ANSI_CLEAR}"
 }
 
+travis_download() {
+  local src="${1}"
+  local dst="${2}"
+
+  if curl --version &>/dev/null; then
+    curl -fsSL -o "${dst}" "${src}" 2>/dev/null
+    return $?
+  fi
+
+  if wget --version &>/dev/null; then
+    wget -q "${src}" -O "${dst}" 2>/dev/null
+    return $?
+  fi
+
+  return 1
+}
+
+travis_wait_for_network() {
+  local result=0
+  local count=1
+  local url="<%= app_host %>/empty.txt"
+  url="${url}?job_id=${TRAVIS_JOB_ID}"
+  url="${url}&repo=${TRAVIS_REPO_SLUG}"
+
+  while [ $count -le 9 ]; do
```

Comment:
Instead of using a custom counter, maybe we can use `curl --retry=9 --retry-delay=1` and `wget --tries=10 --wait=1`.

Reply:
I just read these flags from their man pages. They aren't tested.